### DISCUSSION TITLE
Model's shadow clips sometimes when loading in a new model

### DISCUSSION THREAD

**Discussion by ghost** [google/model-viewer#2736] [Q&A] [answered]
I have an issue where sometime the shadow on a model has a weird glitch where it looks like it has be clipped. This glitch only happens when I reload the model. The image below shows this issue:

![image](https://user-images.githubusercontent.com/7875817/131270633-ca7e81db-7329-4078-848d-7c6d0d8979cd.png)

Here is the demo: https://determined-mirzakhani-828e94.netlify.app/

**Comment by elalish** [comment, marked as answer]
Seems to be working fine for me; can you give specific repro steps?

**Reply by ghost** [reply]
This is my repo for the project -> https://github.com/braedenf/3d-model-viewer/tree/model-viewer-test
On the `model-viewer-test` branch
I'm using Sveltekit for the project.
The bug will seemingly occur at random when I dynamically change the `src` field on the model-viewer.

The issue seems to happens mainly when I have the camera zoomed in when I change, but sometimes it happens anyway. Wish I could provide a better answer.

Try load a different model a number of times and the bug should occur. I've had it happen on android, iOS and pc.

**Reply by elalish** [reply]
Thanks, I have a consistent repro now:
1) Let the first model auto-rotate for a few seconds before interacting
2) Interact to turn the model
3) Change models - now the shadow is wrong. Definitely a bug, thanks!

**Reply by ghost** [reply]
Awesome, glad you could repro, and it wasn't just me 👍

**Reply by elalish** [reply]
Fixed in #2765. Will be in the next release.
